README.md

```diff
@@ -1,2 +1,196 @@
 # Zod to OpenAPI
 
+A library that uses zod schemas to generate an Open API Swagger documentation.
+
+1. [Purpose](#purpose)
+2. [How it works](#how-it-works)
+   1. [Technologies](#technologies)
+   2. [Concept](#concept)
+3. [Quick start](#quick-start) — Fast Track
+   1. [Installation](#installation)
+   2. [Set up](#set-up)
+
+<!-- TODO: Something about a CHANGELOG -->
+
+## Purpose
+
+We at Astea Solutions made this library because of the duplication of work when creating an API validated API with `zod` that needs documentation.
+
+For example:
+
+We can have a zod schema like
+
+```ts
+const UserIdSchema = z.object({
+  id: z.string(),
+});
+const UserSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  age: z.number(),
+});
+```
+
+this schema can be used to validate the response of a `GET /users/{id}` endpoint. So the documentation may look like
+
+```yaml
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: string
+          example: '1212121'
+        name:
+          type: string
+          example: 'John Doe'
+        age:
+          type: number
+          example: 42
+      required:
+        - userId
+        - name
+        - age
+  parameters:
+    UserId:
+      in: header
+      name: id
+      schema:
+        type: string
+      required: true
+      description: The user id to look for
+      allowReserved: true
+      example: '1212121'
+# ... Some other stuff in between
+paths:
```

Comment:
It's a bit confusing to show TS code and then show more than what it generates - users might get the wrong idea. I suggest splitting the example with the schemas and the one with the paths